apps/ejabberd/src/pubsub_migrate.erl

```diff
@@ -0,0 +1,443 @@
+%%%----------------------------------------------------------------------
+%%% File    : pubsub_migrate.erl
+%%% Author  : Christophe Romain <christophe.romain@process-one.net>
+%%% Purpose : Migration/Upgrade code put out of mod_pubsub
+%%% Created : 26 Jul 2014 by Christophe Romain <christophe.romain@process-one.net>
+%%%
+%%%
+%%% ejabberd, Copyright (C) 2002-2015   ProcessOne
+%%%
+%%% This program is free software; you can redistribute it and/or
+%%% modify it under the terms of the GNU General Public License as
+%%% published by the Free Software Foundation; either version 2 of the
+%%% License, or (at your option) any later version.
+%%%
+%%% This program is distributed in the hope that it will be useful,
+%%% but WITHOUT ANY WARRANTY; without even the implied warranty of
+%%% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+%%% General Public License for more details.
+%%%
+%%% You should have received a copy of the GNU General Public License along
+%%% with this program; if not, write to the Free Software Foundation, Inc.,
+%%% 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+%%%
+%%%----------------------------------------------------------------------
+
+-module(pubsub_migrate).
+
+-include("ejabberd.hrl").
+-include("pubsub.hrl").
+
+-export([update_node_database/2, update_state_database/2]).
+-export([update_item_database/2, update_lastitem_database/2]).
+
+update_item_database_binary() ->
```

Comment:
I guess we don't need this function since we start with binaries and there is no need to upgrade from strings.